### ISSUE TITLE
possible bug in Clusterer().cluster(method='gk')

### ISSUE THREAD

**Issue by tdhooghe** [CaroFuchs/pyFUME#4]
Dear Caro,

When I am trying to run the following code:
```
def train_final_model(mf_shape, variable_names, nr_clus):
    all_variable_names = variable_names + ['Price']
    # load data based on selected features
    dl = DataLoader(dataframe=house_data[all_variable_names], normalize='minmax', verbose=False)

    # use all available data from now on to train the model
    cl = Clusterer(nr_clus=nr_clus, x_train=dl.dataX, y_train=dl.dataY, verbose=False)
    _, partition_matrix, _ = cl.cluster(method='gk')  # 'fst-pso' or 'fcm' or 'gk'

    ae = AntecedentEstimator(dl.dataX, partition_matrix)
    antecedent_parameters = ae.determineMF(mf_shape=mf_shape)

    fsc = FireStrengthCalculator(antecedent_parameters, nr_clus=nr_clus, variable_names=variable_names)
    firing_strengths = fsc.calculate_fire_strength(dl.dataX)

    ce = ConsequentEstimator(dl.dataX, dl.dataY, firing_strengths=firing_strengths)
    consequent_parameters = ce.suglms()

    model_builder = SugenoFISBuilder(
        antecedent_parameters,
        consequent_parameters,
        variable_names,
        model_order="first",
        extreme_values=None,
        save_simpful_code=True,
    )
    return model_builder.get_model()


final_model = train_final_model(mf_shape='gauss2', variable_names=variable_names, nr_clus=9)
```
I get the following traceback:
```
Traceback (most recent call last):
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-67-e4ce025dc7cc>", line 30, in <module>
    final_model = train_final_model(mf_shape='gauss2', variable_names=variable_names, nr_clus=2)
  File "<ipython-input-67-e4ce025dc7cc>", line 8, in train_final_model
    _, partition_matrix, _ = cl.cluster(method='gk')  # 'fst-pso' or 'fcm' or 'gk'
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/pyfume/Clustering.py", line 85, in cluster
    centers, partition_matrix, jm =self._gk(m=self.m, max_iter=max_iter)
  File "/home/thomas/Documents/1BM120_computational_intelligence/py39env/lib/python3.9/site-packages/pyfume/Clustering.py", line 352, in _gk
AttributeError: 'Clusterer' object has no attribute 'error'
```
Such a traceback does not occur if the cluster method is set to 'fcm' or 'fst-pso'. Would it be possible to fix this?

Kind regards,
Thomas

**Comment by CaroFuchs** [comment]
Dear Thomas,

Thank you for your message. I tried running your code on my laptop on my machine, and everything seems to be working fine. Which IDE are you using?

Best,

Caro

**Comment by tdhooghe** [comment]
Dear Caro,

Thank you again for your response. It runs now. Maybe something was solved in v1.27 that caused the error in v1.26. That was the version that I was on when the error occurred. Nevertheless, good job! Everything works as expected now.

Kind regards,
Thomas

**Comment by CaroFuchs** [comment]
Good to hear! :)
